pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -787,7 +787,7 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
 
 func setupServiceForEncryption(m *infinispanv1.Infinispan, ser *corev1.Service) {
 	ee := m.Spec.Security.EndpointEncryption
-	if strings.EqualFold(ee.Type, "Service") {
+	if ee != nil && strings.EqualFold(ee.Type, "Service") {
```

Comment:
@rigazilla wdyth about to add getter for `ispn.Spec.Security.EndpointEncryption.Type` including strings.EqualFold conversion, as we have a large number of places to validate it.
Also, good point to align with enum definition maybe in this or separate PR.

Reply:
how about apply Fold conversion in the ApplyDefault?
Or define compare func like IsService()?
Not a big problem cause we're more or less just turning letters uppercase but I don't like getters that do things under the hood.

Reply:
probably fix data in the apply defaults is preferable

Reply:
Yes, agree with you about Fold conversion in the ApplyDefault?
And we can also add function like IsSecurityService for comparison.